src/v/cloud_io/remote.cc

```diff
@@ -250,7 +250,7 @@ ss::future<upload_result> remote::upload_stream(
       content_length);
     std::optional<upload_result> result;
     while (!_gate.is_closed() && permit.is_allowed && !result
-           && max_retries.value_or(1) > 0) {
+           && (!max_retries.has_value() || max_retries.value() > 0)) {
```

Comment:
doesn't this also still have infinite retries -- !max_retries.has_value() will continue (the same as when max_retires.value_or(1) > 0 would continue to be true)?

Reply:
previous it did "look" like there was a mistake, though i couldn't be certain if it was that there should be infinite retries or 1 or none